[DOCS-6326] - Added Java Foundation API page to 5.2

gravitonian opened the pull request.

anxumalo reviewed (commented):
@gravitonian I changed a link to an archived repo and a few minor fixes. 

Can you also check these items:

1. `content-services/5.2/develop/api-reference.md` - this page is missing from the 5.2 Dev Guide TOC.
2. In various pages, should the anchor `#javascript-apijava-api` be just `#javascript-api`?

gravitonian commented:
I have added the api-reference page to the ToC. This is temporary until I have fixed a lot of stuff that are part of DOCS-6346. Changed `#javascript-apijava-api` to `#java-api`

anxumalo reviewed (approved):
LGTM
